go/libkb/globals.go

```diff
@@ -731,6 +739,10 @@ func (g *GlobalContext) Configure(line CommandLine, usage Usage) error {
 	if err := g.ConfigureLogging(); err != nil {
 		return err
 	}
+	if g.Env.GetStandalone() {
+		// If standalone, override the usage to be the same as in a service
+		usage = ServiceUsage
```

Comment:
This doesn't seem right, why do we need to configure the API for doing a `config get`?

Reply:
Because we're running a complete service! In this particular case, the service tries to sync up the configured user and fails because API is not initialized. An alternative fix would be to rework all possible code paths to be aware that some modules might not be initialized.

Reply:
But can we not do that? For example ` echo hello | keybase --standalone base62 encode` works without any API calls. So should config get.

Reply:
Counterpoint: base62 does not interact with the service. It would also crash if it did.

The following commands share this property:
1) base62
2) chat api listen, ctl commands, show-notification - don't make sense in standalone mode, usually explicitly disabled
4) internal kbfs utils - interact with kbfs processes directly
5) installation / configuration utils that only work on osx/windows and interact with the OS.

The following commands will crash in standalone mode:
- everything `config`
- `dismiss pgp-storage`
- some of the things above probably indirectly interact with the service, namely kbfs stuff

Additionally I feel that some commands might run into weird edge cases when we build more and more functionality in the initialization etc. The flow which breaks is the following one:
1) service init triggers d.SetupCriticalSubServices() in service/main.go:329
2) which triggers ephemeral.ServiceInit on line 334 in the same file
3) which then tries to load local user's upak during eklib's initialization

3 possible fixes:
1) add API to Usage of every command that doesn't have it and isn't explicitly disabled in standalone mode
2) make sure that the API is initialized in the eklib initialization code (effectively this will mean that standalone mode will always have an initialized API and there will be another place that runs ConfigureAPI other than ConfigureUsage)
3) this pullreq, which has only a single caveat - some commands that don't interact with the service might end up making client requests because eklib _always_ gets triggered

Reply:
I think that running a `config get` shouldn't trigger anything ephemeral at all, it should work offline. Talking to service is fine but it doesn't need to talk to network. Can we skip `ephemeral.ServiceInit` if in standalone mode? cc @joshblum for ephemeral details

Reply:
let's leave ephemeral in serviceinit since even in standalone expired keys can be deleted

Reply:
So many things do not work in standalone mode.  Nobody uses it outside of keybase.  I don't think we should put any effort into it.

Reply:
That being said, I agree that any form of `keybase config get` should work offline.